src/effects/effectrack.cpp

```diff
@@ -319,21 +323,55 @@ EffectChainSlotPointer PerGroupRack::addEffectChainSlotForGroup(const QString& g
     addEffectChainSlotInternal(pChainSlotPointer);
     m_groupToChainSlot[groupName] = pChainSlotPointer;
 
+    // TODO(rryan): Set up next/prev signals.
+
+    EffectChainPointer pChain(new EffectChain(m_pEffectsManager, chainSlotGroup));
+    pChainSlot->loadEffectChainToSlot(pChain);
+    pChain->addToEngine(getEngineEffectRack(), iChainSlotNumber);
+    // Set the chain to be fully wet.
+    pChain->setMix(1.0);
+    pChain->updateEngineState();
+
     // TODO(rryan): remove.
-    foreach (const ChannelHandleAndGroup& handle_group,
+    const ChannelHandleAndGroup* handleAndGroup;
```

Comment:
We don't need to use a pointer here, because a default constructed ChannelHandleAndGroup() is a well-defined "invalid" value.

How about moving those loops into reusable member functions of EffectChainManager that can be consolidated later?
```
ChannelHandleAndGroup handleAndGroup = m_pEffectChainManager->getInputChannelByGroupName(groupName);
DEBUG_ASSERT(handleAndGroup.valid());
```
This would help to detangle the code step by step and improve readability.

Reply:
I think it would make more sense to make it a function of ChannelHandleFactory.

Reply:
What do you think about renaming ChannelHandleFactory to ChannelHandleManager?